django/db/models/query.py

```diff
@@ -544,19 +544,31 @@ async def aiterator(self, chunk_size=2000):
         An asynchronous iterator over the results from applying this QuerySet
         to the database.
         """
-        if self._prefetch_related_lookups:
-            raise NotSupportedError(
-                "Using QuerySet.aiterator() after prefetch_related() is not supported."
-            )
         if chunk_size <= 0:
             raise ValueError("Chunk size must be strictly positive.")
         use_chunked_fetch = not connections[self.db].settings_dict.get(
             "DISABLE_SERVER_SIDE_CURSORS"
         )
+
+        chunk = []
         async for item in self._iterable_class(
             self, chunked_fetch=use_chunked_fetch, chunk_size=chunk_size
         ):
-            yield item
+            chunk.append(item)
+            if len(chunk) == chunk_size:
+                await sync_to_async(prefetch_related_objects)(
```

Comment:
I guess a sense that we could reuse the `SyncToAsync(prefetch_related_objects)` to avoids creating multiple of them here by defining a `aprefetch_related_objects = sync_to_async(prefetch_related_objects)` outside of the loop and reusing it here and below

```suggestion
                await aprefetch_related_objects(chunk, *self._prefetch_related_lookups)
```